src/core/lombok/eclipse/handlers/HandleNonNull.java

```diff
@@ -73,10 +91,163 @@ public void fix(EclipseNode method) {
 		}
 	}
 	
+	private List<FieldDeclaration> getRecordComponents(EclipseNode typeNode) {
+		List<FieldDeclaration> list = new ArrayList<FieldDeclaration>();
+		
+		for (EclipseNode child : typeNode.down()) {
+			if (child.getKind() == Kind.FIELD) {
+				FieldDeclaration fd = (FieldDeclaration) child.get();
+				if ((fd.modifiers & AccRecord) != 0) list.add(fd);
+			}
+		}
+		
+		return list;
+	}
+	
+	private EclipseNode addCompactConstructorIfNeeded(EclipseNode typeNode, EclipseNode annotationNode) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.